sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/CodeGenInternal/PatchModels/AzureApplicationInsightsDataFeedPatch.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+using Azure.Core;
+
+namespace Azure.AI.MetricsAdvisor.Models
+{
+    internal partial class AzureApplicationInsightsDataFeedPatch
+    {
+        // Full qualification must be used so CodeGen skips the generation of this method.
+        // See: https://github.com/Azure/autorest.csharp/issues/793
+        void global::Azure.Core.IUtf8JsonSerializable.Write(Utf8JsonWriter writer)
+        {
+            writer.WriteStartObject();
+            if (Optional.IsDefined(DataSourceParameter))
+            {
+                writer.WritePropertyName("dataSourceParameter");
+                writer.WriteObjectValue(DataSourceParameter);
+            }
+            SerializeCommonProperties(writer);
```

Comment:
All `<...>DataFeedPatch` models, including this one, are subclasses of `DataFeedDetailPatch`. I put serialization of all properties of the base class in `SerializeCommonProperties` to avoid repeating (a lot of) code, specially since we have 13 subtypes.